### PR TITLE
Linked list merge sort in C, C++, Java, Python #1578

### LINKED ISSUE #1578: Linked List MergeSort in C, C++, Java, Python
I would like to contribute on this issue.

### PR DESCRIPTION
**Pull request by AkanshaKamboj**
Fixes #1578

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- 
- 

#### Languages Used:
- 
- 

#### Files Added:
- 
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Linked_List_Merge_Sort/Linked_List_MergeSort.cpp [hunk 1]
```diff
@@ -0,0 +1,131 @@
+#include<bits/stdc++.h>
```
Comment (MastersAbh) [inline Linked_List_Merge_Sort/Linked_List_MergeSort.cpp]: Please add problem description at top of each file

File: Linked_List_Merge_Sort/Linked_List_MergeSort.cpp [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    
+    if(!l2)
+        return l1;
+
+    struct Node *head=NULL;
+    
+    if(l1->data<l2->data)
+    {
+        head=l1;
+        l1=l1->next;
```
Comment (MastersAbh) [inline Linked_List_Merge_Sort/Linked_List_MergeSort.cpp]: Please give space around operators across the files

File: Linked_List_Merge_Sort/Linked_List_MergeSort.cpp [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    while (head != NULL) {
+        printf("%d-->", head->data);
+        head = head->next;
+    }
+
+    printf("NULL");
+}
+
+/*
+Sample Input
```
Comment (MastersAbh) [inline Linked_List_Merge_Sort/Linked_List_MergeSort.cpp]: Add this sample at the end of file

File: Linked_List_Merge_Sort/Linked_List_MergeSort.java [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+                temp = temp.next;
+                n--;
+            }
+           
+            head = l.mergeSort(head);
+            l.printList(head);
+            t--;
+        }
+    }
+}
```
Comment (MastersAbh) [inline Linked_List_Merge_Sort/Linked_List_MergeSort.java]: Give one blank line at end

**Comment by MastersAbh** [comment]
Please mention the issue number in changelist

**Comment by AkanshaKamboj** [comment]
@MastersAbh I am creating this PR for the same code 4th time. Please tell me all the changes at once as I am unable to work on different issues.

**Comment by asha15** [comment]
please include the issue number next to `Fixes #`

**Comment by AkanshaKamboj** [comment]
@asha15  @MastersAbh  Please review my PR now. I have made all the changes

**Comment by AkanshaKamboj** [comment]
Please review my PR. I need to work on more issues. @asha15 @MastersAbh

**Comment by AkanshaKamboj** [comment]
Please review my PR.

**Comment by AkanshaKamboj** [comment]
@MastersAbh I am not finding any way to squash my commits using web interface. Please help me

**Comment by AkanshaKamboj** [comment]
I saw that there is an option of squash and merge in web interface. If there is any other option please let me know.

**Comment by asha15** [comment]
please reach me on slack

**Comment by MastersAbh** [comment]
@AkanshaKamboj You can squash through git command line, you can reach out on slack to any of us.

**Comment by MastersAbh** [comment]
@AkanshaKamboj are you working on this ?

**Comment by AkanshaKamboj** [comment]
I am unable to do squashing using bash in mac OS. I told it to @asha15 as well.

**Comment by AkanshaKamboj** [comment]
Please help me out. I am not finding any option to squash the commits using web interface. It's showing it could be done only at the time of merge with an option named "squash and merge". Please help me @MasterAbh. I am unable to work on more issues.

**Comment by MastersAbh** [comment]
@AkanshaKamboj It cannot be done through web interface. You need to use git command line for the same. Are you using command line for Git ?

**Comment by AkanshaKamboj** [comment]
No i am not. Actually I am having mac and it needs a software named Xcode for executing git commands using command line but it's off 8gb and needs to be downloaded in one go. I can't do that because of shortage of internet and am not having windows. I really need these codes to be merged.

**Comment by MastersAbh** [comment]
Okay, I will merge then @AkanshaKamboj. But Squashing is important, Also Github desktop is available for Mac, you can try that.

**Pull request merged**